### ISSUE TITLE
Protocol invariance not tracked if subclassed

### ISSUE THREAD

**Issue by Hnasar** [python/mypy#18724]
**Bug Report**

Taking the `Protocol with @property` example: https://mypy.readthedocs.io/en/stable/protocols.html#invariance-of-protocol-attributes
It shows that attributes must be read-only for subclass covariance to work.

However, if a child class explicitly subclasses the `Protocol` then mypy doesn't report the error. This results in unsound code. This seems like a bug in typechecker behavior (and also docs which are confusing.)

**To Reproduce**

```python
class Box(Protocol):
    content: object

def takes_box(box: Box) -> None:
    box.content = object()

@dataclasses.dataclass(slots=False)
class BadBox(Box):  # XXX: inheritence wrongly hides the error
    content: int
```

Full example here: https://mypy-play.net/?mypy=latest&python=3.11&gist=4d7e99ee4f933393234041a219e3f952

**Expected Behavior**

mypy to report that `BadBox.content` doesn't match `Box`, either in the declaration of `content: int` or in the function call to `takes_box`.

**Actual Behavior**

Mypy reports no errors, but at runtime, `BadBox.content` is no longer an `int`.

**Notes**
1. if `Box.content: str`, then a liskov error is reported for `BadBox.content: int`, but because `int` is a subclass of `object` no error seems to be reported.

2. Perhaps `ReadOnly` will help here?  https://peps.python.org/pep-0767/ I tried using `Final` but it's not allowed in `Protocol`s unfortunately.

**Comment by brianschubert** [comment]
This sort of check is implemented by the error code [`mutable-override`](https://mypy.readthedocs.io/en/stable/error_code_list2.html#check-that-overrides-of-mutable-attributes-are-safe-mutable-override) (disabled by default, not implied by `--strict`). If you enable `mutable-override`, you'll get the following error on the declaration of `content: int`:
```
main.py:15: error: Covariant override of a mutable attribute (base class "Box" defined the type as "object", expression has type "int")  [mutable-override]
```

playground: https://mypy-play.net/?mypy=latest&python=3.11&gist=4d7e99ee4f933393234041a219e3f952&enable-error-code=mutable-override
